src/main/java/software/amazon/smithy/gradle/SmithyPlugin.java

```diff
@@ -81,30 +103,62 @@ public void apply(Project project) {
      * the "smithy-model" dependency. If found, the version of smithy-model
      * detected is used for the CLI.
      *
-     * @param project Project to update.
+     * @param project Project to add dependencies to.
      */
     private void addCliDependencies(Project project) {
-        Configuration cli = project.getConfigurations().maybeCreate("smithyCli");
+        Configuration cli;
+        if (project.getConfigurations().findByName("smithyCli") != null) {
+            cli = project.getConfigurations().getByName("smithyCli");
+        } else {
+            cli = project.getConfigurations().create("smithyCli")
+                    .extendsFrom(project.getConfigurations().getByName("runtimeClasspath"));
+        }
+
+        // Prefer explicitly set dependency first.
         DependencySet existing = cli.getAllDependencies();
 
         if (existing.stream().anyMatch(d -> isMatchingDependency(d, "smithy-cli"))) {
-            LOGGER.info("Using explicitly configured Smithy CLI");
+            project.getLogger().warn("(using explicitly configured Smithy CLI)");
             return;
         }
 
-        String cliVersion = project.getConfigurations().getByName("runtimeClasspath").getAllDependencies().stream()
+        failIfRunningInMainSmithyRepo(project);
+        String cliVersion = detectCliVersionInDependencies(SmithyUtils.getClasspath(project, "runtimeClasspath"));
```

Comment:
Note that this doesn't do a transitive dependency check. Doing so is apparently super slow, so I avoided it for now. We can look to do that in the future, but I'd want to use some kind of caching, ideally provided by Gradle itself.